core/slice.go

```diff
@@ -513,12 +513,21 @@ func (sl *Slice) SubRelayPendingHeader(pendingHeader types.PendingHeader, reorg
 	defer sl.phCachemu.Unlock()
 
 	// Collect our manifest, so that the dom header can commit to it
-	if nodeCtx > common.PRIME_CTX {
-		manifest, err := sl.hc.CollectBlockManifest(pendingHeader.Header)
-		if err != nil {
-			log.Warn("Failed to get manifest for pending header", "parentHash: ", pendingHeader.Header.ParentHash(), "err: ", err)
+	if nodeCtx > common.PRIME_CTX && pendingHeader.Header.NumberU64() > 0 {
```

Comment:
That's precisely why I designed it this way. The manifest is collected each time by traversing the set of ancestors for this block, rather than keeping some state variable which could become stale when appending a block to a different head.